### PR TITLE
Wordpress fix

### PR DESCRIPTION
**Pull request by rohanagarwal94**
Fixes issue #53

Changes: A lot of changes have been made in this particular pull request reflecting the work of the past few days. The majority of the code has been taken help from Wordpress-android which is open-sourced. Due to poor documentation(personal opinion) of the API endpoints by wordpress for developer I rather used the networking library used by Wordpress-android itself to make the necessary API calls refer [this](https://github.com/wordpress-mobile/WordPress-FluxC-Android/wiki/Using-FluxC) and [this](https://github.com/wordpress-mobile/WordPress-FluxC-Android). Suggestions welcome.

Screenshots for the change: 
screenshot_2017-04-14-01-23-50-349_vn mbm phimp me
screenshot_2017-04-14-01-23-42-014_vn mbm phimp me
screenshot_2017-04-14-01-23-09-411_vn mbm phimp me
screenshot_2017-04-14-01-23-02-963_vn mbm phimp me
screenshot_2017-04-14-01-22-38-785_vn mbm phimp me
screenshot_2017-04-14-01-22-31-882_vn mbm phimp me

### CODE REVIEW

**Comment by phimpme-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">:tada: Thanks for the PR @rohanagarwal94 we hope you have changed the variables in the upload_apk_forks.sh its for easy moderation.</td>
    </tr>
  </tbody>
</table>

:link: Maintainers: the apk to test can be found at https://github.com/rohanagarwal94/phimpme-android/blob/apk/app-debug.apk

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by mariobehling** [comment]
Thanks a lot. This is a very big change.

Currently Travis is failing. Could you resolve this, please? Cheers.

**Comment by rohanagarwal94** [comment]
@mariobehling I am currently solving that only. Cheers.

File: .travis.yml [hunk 1]
```diff
@@ -53,6 +53,7 @@ before_script:
   - echo no | android create avd --force -n test -t android-19 --abi armeabi-v7a
   - emulator -avd test -no-audio -no-window &
   - sleep 300
+  - android-wait-for-emulator
```
Comment (Abhi2424shek) [inline .travis.yml]: sleep is used instead, android-wait-for-emulator is still buggy and can result in test failures, adding both will work but will waste time.

**Comment by rohanagarwal94** [comment]
@pa1pal @Abhi2424shek I ran the failing test in my local machine, and it ran perfectly. Could you help me with this so that we can speed up the development. Thanks.

File: .travis.yml (hunk 1 above)
Comment (rohanagarwal94) [inline .travis.yml]: It is failing in either case @Abhi2424shek

**Comment by pa1pal** [comment]
@rohanagarwal94 how did you run travis locally? using docker image?

File: .travis.yml (hunk 1 above)
Comment (Abhi2424shek) [inline .travis.yml]: It's not about test failures, its that this line is unwanted.

**Comment by Abhi2424shek** [comment]
@rohanagarwal94,
>com.android.builder.testing.ConnectedDevice > No tests found.[test(AVD) - 4.4.4] FAILED 
No tests found.
This usually means that your test classes are not in the form that your test runner expects(e.g. don't inherit from TestCase or lack @Test annotations).
:app:connectedDebugAndroidTest FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':app:connectedDebugAndroidTest'.

Can you try not to compile dagger and use `apt` it may fix the tests. Please the gradle file.
Hope it helps.

**Comment by rohanagarwal94** [comment]
@pa1pal i ran `./gradlew assembleDebug --stacktrace` and it ran successfully. @Abhi2424shek just reverting the last commit. Dagger is being used for dependency injection which is a core component of the fluxc networking library so therefore cant really compile the project without using it.

**Comment by mariobehling** [comment]
@rohanagarwal94 Thank you. Please check travis logs. For the future please make smaller commits and make it easier for reviewers.

**Comment by Abhi2424shek** [comment]
As I mentioned @rohanagarwal it's a bug with the Android Test Runner, maybe you can try to compile another version of dagger? This generally occurs when the inbuilt version of dagger comes into conflict with the one you are compiling into your app.

**Comment by pa1pal** [comment]
@rohanagarwal94  The `apt` function was included in the latest android gradle plugin. It's called `annotationProcessor`. Use `annotationProcessor` instead of `apt` for dagger. It might work.

**Comment by rohanagarwal94** [comment]
@mariobehling I will definitely remember that in future. Its just that this time the priority of the issue overshadowed everything. @pa1pal I think [this](https://stackoverflow.com/questions/26939340/how-do-you-override-a-module-dependency-in-a-unit-test-with-dagger-2-0?lq=1) might be the possible solution since I discovered that I did change the upload.java file and the espresso tests need to be updated too of the same class. Meanwhile , I am also trying your workaround.

**Comment by Abhi2424shek** [comment]
The tests are not being detected @rohanagarwal94

**Comment by rohanagarwal94** [comment]
@pa1pal changed to annotation processor, @Abhi2424shek updated dagger to latest version. Unfortunately, same output. For reference, see [this](https://travis-ci.org/rohanagarwal94/phimpme-android)

**Comment by Abhi2424shek** [comment]
Try using the same version as used in the Android Test Runner.

**Pull request closed without merging**